heron/executor/src/python/heron_executor.py

```diff
@@ -610,24 +619,37 @@ def _untar_if_tar(self):
   # pylint: disable=no-self-use
   def _wait_process_std_out_err(self, name, process):
     ''' Wait for the termination of a process and log its stdout & stderr '''
-    (process_stdout, process_stderr) = process.communicate()
-    if process_stdout:
-      Log.info("%s stdout: %s" %(name, process_stdout))
-    if process_stderr:
-      Log.info("%s stderr: %s" %(name, process_stderr))
+    log.stream_process_stdout(process, stdout_log_fn(name))
+    process.wait()
 
   def _run_process(self, name, cmd, env_to_exec=None):
     Log.info("Running %s process as %s" % (name, ' '.join(cmd)))
-    return subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                            env=env_to_exec)
+    try:
+      # stderr is redirected to stdout so that it can more easily be logged. stderr has a max buffer
+      # size and can cause the child process to deadlock if it fills up
+      process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+                                 env=env_to_exec)
```

Comment:
How about adding a [bufsize=1](https://github.com/twitter/heron/blob/master/heron/tools/cli/src/python/execute.py#L71) here for better performance?